capytaine/bodies/bodies.py

```diff
@@ -270,6 +270,27 @@ def assemble_regular_array(self, distance, nb_bodies):
                 array_dofs[f'{i}_{j}__{dof_name}'] = new_dof
         return FloatingBody(mesh=array_mesh, dofs=array_dofs, name=f"array_of_{self.name}")
 
+    def assemble_arbitrary_array(self, locations:np.ndarray):
+
+        if not isinstance(locations, np.ndarray):
+            raise TypeError('locations must be of type np.ndarray')
+        assert locations.shape[1] == 2, 'locations must be of shape nx2, received {:}'.format(locations.shape)
+        n = locations.shape[0]
+
+        fb_list = []
+        for idx, li in enumerate(locations):
+            fb1 = self.copy()
+            fb1.translate(np.append(li,0))
+            fb1.name = 'arbitrary_array_body{:02d}'.format(idx)
+            for dof_key in list(self.dofs.keys()):
+                new_dof_key = fb1.name + '_' + dof_key
+                fb1.dofs[new_dof_key] = fb1.dofs.pop(dof_key)
```

Comment:
Renaming the dofs is already done by `FloatingBody.join_bodies`, isn't it?

Reply:
right! b3a6304c8f969076101d17a1b451e56474d39698